### PR TITLE
fix: Remove absolute path in config files

### PR DESCRIPTION
**Pull request by sposadac**
### Description

I am proposing a minor change, namely to ommit appending the absolute path to the config file. This is not compatible with cloud execution (at least google life sciences), when using `--configfile` command line option. 

### QC

* [x] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [x] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

**Comment by cademirch** [comment]
Support this - I noticed this in #1833 but never got around to making a PR. Thanks for opening one!

File: snakemake/__init__.py [hunk 1]
```diff
@@ -498,8 +498,6 @@ def snakemake(
     for f in configfiles:
         # get values to override. Later configfiles override earlier ones.
         overwrite_config.update(load_configfile(f))
-    # convert provided paths to absolute paths
-    configfiles = list(map(os.path.abspath, configfiles))
```
[the remark this answers is not in the document]
Comment (cademirch) [reply snakemake/__init__.py:502]: Perhaps in the cluster/cloud executor we can coerce the path back to relative to the workflow?

[the remark this answers is not in the document]
Comment (johanneskoester) [reply snakemake/__init__.py:502]: Yes, that is probably the better choice.

**Comment by sposadac** [comment]
> @sposadac do you think you can resolve the conflicts. It seems that on snakemake8 that code was moved to:
> 
> https://github.com/snakemake/snakemake/blob/74627d3f07f549600dc8ad226a545f44191141e0/snakemake/settings.py#L328-L329

I believe it still doesn't work when using cloud executors because this function is called in the `__post_init__` method as an additional initialization task, and the absolute path in the current directory is prepended. I also think this happens before the executor run, so the issue will remain, but I need to verify that.

**Comment by johanneskoester** [comment]
Thanks for bringing this up! I have found a different solution to this, which is more in line with the changes that come with Snakemake 8: #3240

**Pull request closed without merging**